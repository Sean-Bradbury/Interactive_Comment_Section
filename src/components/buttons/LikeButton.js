import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

//Colors
import colors from '../../styles/colors';

//images
import PlusImage from '../../images/icon-plus.svg';
import MinusImage from '../../images/icon-minus.svg';

const LikeButtonContainer = styled.div`
    display: flex;
    align-items: center;
    padding: 0.5rem 1rem;
    border-radius: 10px;
    background-color: ${colors.colorNeutralVLightGray};
    &.desktop {
        flex-direction: column;
        align-items: center;
        padding: 1rem 0.5rem;
    }
`;

const Score = styled.span`
    padding: 0 1rem;
    color: ${colors.colorPrimaryModBlue};
    font-weight: 500;
    &.desktop {
        padding: 1rem 0;
    }
`;

const LikeButtonImage = styled.img`
    width: 10px;
`;

const LikeButton = ({score, className}) => {
  return (
    <LikeButtonContainer className={className}>
        <LikeButtonImage src={PlusImage} alt="Plus" />
        <Score className={className}>
            {score}
        </Score>
        <LikeButtonImage src={MinusImage} alt="Minus" />
    </LikeButtonContainer>
  )
}

LikeButton.propTypes = {
    score: PropTypes.string.isRequired,
    className: PropTypes.string.isRequired,
};

export default LikeButton;
