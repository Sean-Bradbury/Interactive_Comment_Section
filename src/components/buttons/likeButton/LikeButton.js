import { useState, useContext, useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

//Colors
import colors from '../../../styles/colors';

//images
import PlusImage from '../../../images/icon-plus.svg';
import MinusImage from '../../../images/icon-minus.svg';

//context
import commentContext from '../../../context/comment/commentContext';

const LikeButtonContainer = styled.div`
  display: flex;
  align-items: center;
  align-self: start;
  justify-content: space-between;
  padding: 0.5rem 1rem;
  border-radius: 10px;
  background-color: ${colors.colorNeutralVLightGray};
  &.desktop {
    flex-direction: column;
    align-items: center;
    padding: 1rem 0;
  }
`;

const Score = styled.span`
  padding: 0 0.5rem;
  color: ${colors.colorPrimaryModBlue};
  font-weight: 500;
  min-width: 37px;
  &.desktop {
    padding: 1rem 0;
  }
`;

const LikeButtonImage = styled.img`
  object-fit: contain;
  object-position: center;
  padding: 5px;
  cursor: pointer;
  &:hover {
    filter: brightness(50%);
  }
  &:active,
  &:focus {
    opacity: 0.5;
  }
`;

const LikeButton = ({ score, className, id }) => {
  const [currentScore, setCurrentScore] = useState(score);

  const CommentContext = useContext(commentContext);
  const { updateLikesComment } = CommentContext;

  useEffect(() => {
    updateLikesComment(id, currentScore);
    // eslint-disable-next-line
  }, [currentScore]);

  const handleScoreClick = (direction) => {
    if (direction === 'plus') {
      setCurrentScore(currentScore + 1);
    } else if (direction === 'minus') {
      if (currentScore > 0) {
        setCurrentScore(currentScore - 1);
      }
    }
  };

  return (
    <LikeButtonContainer
      data-testid="like-button"
      className={className}
    >
      <LikeButtonImage
        src={PlusImage}
        alt="Plus"
        onClick={() => handleScoreClick('plus')}
      />
      <Score className={className}>{currentScore}</Score>
      <LikeButtonImage
        src={MinusImage}
        alt="Minus"
        onClick={() => handleScoreClick('minus')}
      />
    </LikeButtonContainer>
  );
};

LikeButton.propTypes = {
  score: PropTypes.number.isRequired,
  className: PropTypes.string.isRequired,
  id: PropTypes.any,
};

export default LikeButton;
