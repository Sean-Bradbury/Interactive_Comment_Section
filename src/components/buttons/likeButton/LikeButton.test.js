import { cleanup, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LikeButton from './LikeButton';

afterEach(cleanup);

test('Should render component with score of 6', () => {
  render(<LikeButton score={6} className="desktop" />);
  const likeButtonEl = screen.getByTestId('like-button');
  expect(likeButtonEl).toBeInTheDocument();
  expect(likeButtonEl).toHaveTextContent('6');
});

test('Should render component with score of 10', () => {
  render(<LikeButton score={10} className="desktop" />);
  const likeButtonEl = screen.getByTestId('like-button');
  expect(likeButtonEl).toBeInTheDocument();
  expect(likeButtonEl).toHaveTextContent('10');
});

test('matches snapshot', () => {
  const tree = renderer
    .create(<LikeButton score={6} className="desktop" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
