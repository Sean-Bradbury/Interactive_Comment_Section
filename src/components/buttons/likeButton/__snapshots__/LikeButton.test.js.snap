// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bczRLJ ivHeaC desktop"
  data-testid="like-button"
>
  <img
    alt="Plus"
    className="sc-dkzDqf hGjhtN"
    id="like-button-plus"
    onClick={[Function]}
    src="icon-plus.svg"
  />
  <span
    className="sc-gsnTZi IwjZD desktop"
  >
    6
  </span>
  <img
    alt="Minus"
    className="sc-dkzDqf hGjhtN"
    id="like-button-minus"
    onClick={[Function]}
    src="icon-minus.svg"
  />
</div>
`;
